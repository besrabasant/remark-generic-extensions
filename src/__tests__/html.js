import test from "ava"

import { transformToHtml } from "./helpers/transform"

test(
  "should add an extension",
  t => t.snapshot(transformToHtml("!icon"))
)

test(
  "should add an extension with an id",
  t => t.snapshot(transformToHtml("!icon{ #my-id }"))
)

test(
  "should add an extension with a class",
  t => t.snapshot(transformToHtml("!icon{ .my-class }"))
)

test(
  "should add an extension with an attribute",
  t => t.snapshot(transformToHtml("!icon{ attr=my-attr }"))
)

test(
  "should add an extension with a quoted attribute",
  t => t.snapshot(transformToHtml(`!icon{ attr="my attribute" }`))
)

test(
  "should add an extension with a lone attribute",
  t => t.snapshot(transformToHtml(`!icon{ attr }`))
)

test(
  "should add an extension with a lone attribute with a default value",
  t => t.snapshot(
    transformToHtml(
      `!icon{ attr }`,
      {
        elements: {
          icon: {
            propsDefaultValues: {
              attr: "default",
            }
          }
        }
      }
    )
  )
)

test(
  "should add an extension with all types of parameters",
  t => t.snapshot(
    transformToHtml(
      `!icon{ #my-id .my-class attr attr2=my-attr attr3="my attribute" }`,
      {
        elements: {
          icon: {
            propsDefaultValues: {
              attr: "default",
            }
          }
        }
      }
    )
  )
)

test(
  "should add an extension with a mapped content",
  t => t.snapshot(
    transformToHtml(
      "!icon[my-tooltip]",
      {
        elements: {
          icon: {
            html: {
              tooltip: "::content::"
            }
          }
        }
      }
    )
  )
)

test(
  "should add an extension with a mapped argument",
  t => t.snapshot(
    transformToHtml(
      "!icon(my-icon)",
      {
        elements: {
          icon: {
            html: {
              icon: "::argument::"
            }
          }
        }
      }
    )
  )
)

test(
  "should work with a different placeholder",
  t => t.snapshot(
    transformToHtml(
      "!icon(my-icon)",
      {
        placeholder: "||",
        elements: {
          icon: {
            html: {
              icon: "||argument||"
            }
          }
        }
      }
    )
  )
)

test(
  "should add an extension with a replaced tag",
  t => t.snapshot(
    transformToHtml(
      "!icon",
      {
        elements: {
          icon: {
            html: {
              tagName: "my-icon"
            }
          }
        }
      }
    )
  )
)

test(
  "should work with other markdown",
  t => t.snapshot(
    transformToHtml(
      `#heading

!icon`
    )
  )
)

test(
  "should work with an element with multiple children",
  t => t.snapshot(
    transformToHtml(
      "!icon",
      {
        elements: {
          icon: {
            html: {
              children: [
                {
                  tagName: "p",
                  children: [
                    {
                      type: "text",
                      value: "test"
                    }
                  ]
                }
              ]
            }
          }
        }
      }
    )
  )
)

test(
  "should work with an element with a mapped property on a child",
  t => t.snapshot(
    transformToHtml(
      "!icon{ #test }",
      {
        elements: {
          icon: {
            html: {
              children: [
                {
                  tagName: "p",
                  id: "::prop::id::"
                }
              ]
            }
          }
        }
      }
    )
  )
)

test(
  "should work with an element with multiple children on the same level",
  t => t.snapshot(
    transformToHtml(
      "!icon",
      {
        elements: {
          icon: {
            html: {
              children: [
                {
                  tagName: "p",
                  children: [
                    {
                      type: "text",
                      value: "test"
                    }
                  ]
                },
                {
                  tagName: "span",
                  children: [
                    {
                      type: "text",
                      value: "bla"
                    }
                  ]
                }
              ]
            }
          }
        }
      }
    )
  )
)
