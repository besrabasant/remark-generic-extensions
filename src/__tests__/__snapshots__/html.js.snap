// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should add an extension 1`] = `
"<p><icon></icon></p>
"
`;

exports[`should add an extension with a class 1`] = `
"<p><icon class=\\"my-class\\"></icon></p>
"
`;

exports[`should add an extension with a lone attribute 1`] = `
"<p><icon attr=\\"\\"></icon></p>
"
`;

exports[`should add an extension with a lone attribute with a default value 1`] = `
"<p><icon attr=\\"default\\"></icon></p>
"
`;

exports[`should add an extension with a mapped argument 1`] = `
"<p><icon icon=\\"my-icon\\"></icon></p>
"
`;

exports[`should add an extension with a mapped content 1`] = `
"<p><icon tooltip=\\"my-tooltip\\"></icon></p>
"
`;

exports[`should add an extension with a quoted attribute 1`] = `
"<p><icon attr=\\"my attribute\\"></icon></p>
"
`;

exports[`should add an extension with a replaced tag 1`] = `
"<p><my-icon></my-icon></p>
"
`;

exports[`should add an extension with all types of parameters 1`] = `
"<p><icon id=\\"my-id\\" class=\\"my-class\\" attr=\\"default\\" attr2=\\"my-attr\\" attr3=\\"my attribute\\"></icon></p>
"
`;

exports[`should add an extension with an attribute 1`] = `
"<p><icon attr=\\"my-attr\\"></icon></p>
"
`;

exports[`should add an extension with an id 1`] = `
"<p><icon id=\\"my-id\\"></icon></p>
"
`;

exports[`should work with a different placeholder 1`] = `
"<p><icon icon=\\"my-icon\\"></icon></p>
"
`;

exports[`should work with an element with a mapped property on a child 1`] = `
"<p><icon><p id=\\"test\\"></p></icon></p>
"
`;

exports[`should work with an element with multiple children 1`] = `
"<p><icon><p>test</p></icon></p>
"
`;

exports[`should work with an element with multiple children on the same level 1`] = `
"<p><icon><p>test</p><span>bla</span></icon></p>
"
`;

exports[`should work with other markdown 1`] = `
"<p>#heading</p>
<p><icon></icon></p>
"
`;
